src/gridstack.js

```diff
@@ -1041,10 +1041,14 @@
         this._updateContainerHeight();
     };
 
-    GridStack.prototype.destroy = function() {
+    GridStack.prototype.destroy = function(detachGrid) {
         $(window).off('resize', this.onResizeHandler);
         this.disable();
-        this.container.remove();
+        if (typeof detachGrid != 'undefined' && !detachGrid) {
+            this.removeAll(true);
```

Comment:
You're absolutely correct. This should be `this.removeAll(false)`. We're in the process of adding in unit tests, and clearly have not added this yet. Apologies for the mixup - I'll fix this and have it pushed out momentarily. Thanks for finding and reporting this bug - saved someone lots of frustration down the road. I hope this didn't cause you much issue.